firstvoices/backend/tests/test_tasks/test_import_job_tasks.py

```diff
@@ -740,14 +731,211 @@ def test_duplicate_media_filenames(self):
+            data=file,
+            validation_status=JobStatus.ACCEPTED,
+        )
+        validate_import_job(import_job.id)
+
+        import_job = ImportJob.objects.get(id=import_job.id)
+        validation_report = import_job.validation_report
+        assert validation_report.error_rows == 0
+        assert validation_report.new_rows == 4
+
+    def test_related_media_id_success_shared_site(self):
+        audio = factories.AudioFactory.create(id="7763ae50-1b6e-46bc-bd3a-91037ac736cb")
+        factories.SiteFeatureFactory.create(
+            site=audio.site, key="shared_media", is_enabled=True
+        )
+
+        image = factories.ImageFactory.create(id="daf8e74f-f20b-4c81-95c2-7dd744277009")
+        factories.SiteFeatureFactory.create(
+            site=image.site, key="shared_media", is_enabled=True
+        )
+
+        video = factories.VideoFactory.create(id="4764e764-7830-4bea-b30e-4e35cc93b12b")
+        factories.SiteFeatureFactory.create(
+            site=video.site, key="SHARED_MEDIA", is_enabled=True
+        )
 
+        file_content = get_sample_file(
+            "import_job/minimal_media_ids.csv", self.MIMETYPE
+        )
+        file = factories.FileFactory(content=file_content)
+        import_job = factories.ImportJobFactory(
+            site=self.site,
+            run_as_user=self.user,
+            data=file,
+            validation_status=JobStatus.ACCEPTED,
+        )
         validate_import_job(import_job.id)
 
         import_job = ImportJob.objects.get(id=import_job.id)
         validation_report = import_job.validation_report
         assert validation_report.error_rows == 0
+        assert validation_report.new_rows == 4
+
+    def test_related_media_id_failed_shared_site(self):
+        audio = factories.AudioFactory.create(id="7763ae50-1b6e-46bc-bd3a-91037ac736cb")
+        factories.SiteFeatureFactory.create(
+            site=audio.site, key="shared_media", is_enabled=False
+        )
+
+        image = factories.ImageFactory.create(id="daf8e74f-f20b-4c81-95c2-7dd744277009")
+        factories.SiteFeatureFactory.create(
+            site=image.site, key="shared_media", is_enabled=False
+        )
+
+        video = factories.VideoFactory.create(id="4764e764-7830-4bea-b30e-4e35cc93b12b")
+        factories.SiteFeatureFactory.create(
+            site=video.site, key="NOT_SHARED_MEDIA", is_enabled=True
+        )
+
+        file_content = get_sample_file(
+            "import_job/minimal_media_ids.csv", self.MIMETYPE
+        )
+        file = factories.FileFactory(content=file_content)
+        import_job = factories.ImportJobFactory(
+            site=self.site,
+            run_as_user=self.user,
+            data=file,
+            validation_status=JobStatus.ACCEPTED,
+        )
+        validate_import_job(import_job.id)
+
+        import_job = ImportJob.objects.get(id=import_job.id)
+        validation_report = import_job.validation_report
+        assert validation_report.error_rows == 4
```

Comment:
Initially this test also had an assertion for validation_report.new_rows == 0, but I believe in this case the entries would be imported just without the related media. That is the expected behaviour yes?